lib/msal-core/samples/react-sample-app/src/AuthProvider.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from "react";
+import {
+    msalApp,
+    requiresInteraction,
+    fetchMsGraph,
+    isIE,
+    GRAPH_ENDPOINTS,
+    GRAPH_SCOPES,
+    GRAPH_REQUESTS
+} from "./auth-utils";
+
+// If you support IE, our recommendation is that you sign-in using Redirect APIs
+const useRedirectFlow = isIE();
+
+export default C =>
+    class AuthProvider extends Component {
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                account: null,
+                error: null,
+                emailMessages: null,
+                graphProfile: null
+            };
+        }
+
+        async acquireToken(request, redirect) {
+            return msalApp.acquireTokenSilent(request).catch(error => {
+                // Call acquireTokenPopup (popup window) in case of acquireTokenSilent failure
+                // due to consent or interaction required ONLY
+                if (requiresInteraction(error.errorCode)) {
+                    return redirect
+                        ? msalApp.acquireTokenRedirect(request)
+                        : msalApp.acquireTokenPopup(request);
+                }
+            });
+        }
+
+        async onSignIn(redirect) {
+            if (redirect) {
+                return msalApp.loginRedirect(GRAPH_REQUESTS.LOGIN);
+            }
+
+            const loginResponse = await msalApp
+                .loginPopup(GRAPH_REQUESTS.LOGIN)
+                .catch(() => {
```

Comment:
do you want to define the err here and log / set error

```js
...
.catch(error=> {
    this.setState({
          error // or maybe construct a message from it 
    })
})
```

Reply:
For this error looks like I could do that yeah.

Reply:
(that isn't the case with all the error scenarios, which is why I wasn't doing so originally).